onadata/apps/api/management/commands/delete_users.py

```diff
@@ -0,0 +1,102 @@
+    user_projects = Project.objects.filter(created_by=user).count()
+    # Get the number of forms
+    user_forms = XForm.objects.filter(user=user).count()
+    # Get the number of submissions
+    user_sumbissions = Instance.objects.filter(user=user).count()
+
+    result = {
+        'projects': user_projects,
+        'forms': user_forms,
+        'submissions': user_sumbissions
+    }
+
+    return result
+
+
+def inactivate_user(username):
+    """
+    Soft deletes the user termporarily.
+    """
+    try:
+        user = User.objects.get(username=username)
+        # set inactive status on user account
+        user.is_active = False
+        # append a timestamped suffix to the username
+        # to make the initial username available
+        soft_deletion_time = timezone.now()
+        deletion_suffix = soft_deletion_time.strftime('-deleted-at-%s')
+        user.username += deletion_suffix
+
+        user.save()
+        sys.stdout.write(
+            sys.style.SUCCESS(
+                'User {} deleted successfully.'.format(username)))
+        # confirm too that no user exists with provided email
+        if len(User.objects.filter(email=user.email, is_active=True)) > 1:
+            inactivate_user(user.username)
+    except User.DoesNotExist:
+        sys.stdout.write('User {} does not exist.'.format(username))
+
+
+class Command(BaseCommand):
+    """
+    Delete users management command.
+    """
+    help = 'Delete users'
+
+    def add_arguments(self, parser):
+        parser.add_argument('--user-details', nargs='*')
+
+        parser.add_argument(
+            '--user_input',
+            help='Confirm deletion of user account',
+            default=False
+        )
+
+    def handle(self, *args, **kwargs):
+        users = kwargs.get('user')
+        user_input = kwargs.get('user_input')
+
+        if users:
+            for user in users:
+                username, email = user.split(':')
+                user_stats = get_user_object_stats(username)
+                if user_input is False:
+                    # If the --user_input flag is not provided.
+                    # Get acknowledgement from the user on this
+                    user_response = input(
+                        "User account '{}' has {} projects, {} forms and {} submissions. \
+                        Do you wish to continue deleting this account?".format(
+                            username,
+                            user_stats['projects'],
+                            user_stats['forms'],
+                            user_stats['submissions']
+                            ))
+                    if user_response is True:
+                        inactivate_user(username)
+                        sys.stdout.write(
+                            "User Account {} deleted successfully!".format(
+                                username))
+                elif user_input is True:
+                    inactivate_user(username)
```

Comment:
```suggestion
                if user_input:
                    inactivate_user(username)
                else:
                    # If the --user_input flag is not provided.
                    # Get acknowledgement from the user on this
                    user_response = input(
                        "User account '{}' has {} projects, {} forms and {} submissions. \
                        Do you wish to continue deleting this account?".format(
                            username,
                            user_stats['projects'],
                            user_stats['forms'],
                            user_stats['submissions']
                            ))
                    if user_response:
                        inactivate_user(username)
```

Reply:
I'm curious what's the main purpose of the `user_input` flag ? What is it supposed to trigger in the command ?

Reply:
If it's supposed to trigger the prompt to confirm deletion... I believe the prompt should pop up if `user_input` is `True`

Reply:
You could also possibly shorten this too something like:

```
delete_user = True   # user_response is probably a better name ? not sure

if user_input:
      # If the suggest on the user_input flag being the trigger for the prompt is fine
      # Then we prompt here
      delete_user = prompt here for to reset 'delete_user'

if user_response:
      invalidate_user

Reply:
> If it's supposed to trigger the prompt to confirm deletion...

Yes, it is meant to prompt the user running the command to confirm that they would like to delete that user.

> I believe the prompt should pop up if user_input is True

I think its the other way round...
If a user does not provide this when running the management command i.e
`./manage.py delete_users --user winny:kiraguwinnie@gmail.com wkiragu:wkiragu@ona.io --settings=onadata.settings.local_settings`
then in this case, the default value for user_input is false, and with that, the command prompts the user for confirmation.

If it is provided when running the command then it doesnt need to prompt for confirmation i.e
`./manage.py delete_users --user_details winny:kiraguwinnie@gmail.com wkiragu:wkiragu@ona.io --user_input True --settings=onadata.settings.local_settings`

For cases when the user just needs to delete the user accounts, no information needed.

Reply:
Yea... My issue with this is that the naming and the value clashes a bit.... I guess it's more clear if you make the variable a question like `User Input ?`... Kinda asks if there should be any user input during the process